modules/katello_repository_set.py

```diff
@@ -0,0 +1,188 @@
+            - Organization name that the lifecycle environment is in
+        required: true
+    state:
+        description:
+            - Whether the repositories are enabled or not
+        required: true
+        choices:
+            - 'enabled'
+            - 'disabled'
+'''
+
+EXAMPLES = '''
+- name: "Enable RHEL 7 RPMs repositories"
+  katello_lifecycle_environment:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://katello.example.com"
+    verify_ssl: false
+    name: "Red Hat Enterprise Linux 7 Server (RPMs)"
+    organization: "Default Organization"
+    product: "Red Hat Enterprise Linux Server"
+    repositories:
+    - releasever: "7.0"
+      basearch: "x86_64"
+    - releasever: "7.1"
+      basearch: "x86_64"
+    - releasever: "7.2"
+      basearch: "x86_64"
+    - releasever: "7.3"
+      basearch: "x86_64"
+    state: enabled
+'''
+
+RETURN = '''# '''
+
+try:
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        find_organization,
+        find_product,
+        find_repository_set,
+        ping_server,
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+
+def get_desired_repos(desired_substitutions, available_repos):
+    desired_repos = []
+    for sub in desired_substitutions:
+        desired_repos += filter(lambda available: available['substitutions'] == sub, available_repos)
+    return desired_repos
+
+
+def repository_set(module, name, organization, product, state, repositories=[]):
+    changed = False
+    organization = find_organization(module, organization)
+    product = find_product(module, product, organization)
+    repo_set = find_repository_set(module, name=name, product=product)
+
+    available_repos = repo_set.available_repositories()['results']
+    current_repos = map(lambda r: r.read(), repo_set.repositories)
+    desired_repos = get_desired_repos(repositories, available_repos)
+
+    available_repo_names = set(map(lambda r: r['repo_name'], available_repos))
+    current_repo_names = set(map(lambda r: r.name, current_repos))
+    desired_repo_names = set(map(lambda r: r['repo_name'], desired_repos))
+
+    if len(desired_repo_names - available_repo_names) > 0:
+        module.fail_json(msg="Desired repositories are not available on the repository set {}. Desired: {} Available: {}"
+                         .format(name, desired_repo_names, available_repo_names))
+
+    if state == 'enabled':
+        for repo in desired_repo_names - current_repo_names:
+            repo_to_enable = (r for r in available_repos if r['repo_name'] == repo).next()
+            repo_set.enable(data=repo_to_enable['substitutions'])
+            changed = True
+    else:
+        for repo in current_repo_names & desired_repo_names:
+            repo_to_disable = (r for r in available_repos if r['repo_name'] == repo).next()
+            repo_set.disable(data=repo_to_disable['substitutions'])
+            changed = True
```

Comment:
What happens if it tries to disable a repository set that cannot be disabled due to being in a content view already? Will Nailgun propagate a nice error to the user in that case?